File: src/main/java/com/refinedmods/refinedstorage/item/FluidStoragePartItem.java
package com.refinedmods.refinedstorage.item;

import com.refinedmods.refinedstorage.RS;
import com.refinedmods.refinedstorage.RSItems;
import com.refinedmods.refinedstorage.apiimpl.storage.FluidStorageType;
import net.minecraft.world.item.Item;

public class FluidStoragePartItem extends Item {
    public FluidStoragePartItem() {
        super(new Item.Properties().tab(RS.CREATIVE_TAB));
    }

    public static FluidStoragePartItem getByType(FluidStorageType type) {
        return RSItems.FLUID_STORAGE_PARTS.get(type).get();
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/item/FluidStoragePartItem.kt
package com.refinedmods.refinedstorage.item

import com.refinedmods.refinedstorage.RS
import com.refinedmods.refinedstorage.RSItems
import com.refinedmods.refinedstorage.apiimpl.storage.FluidStorageType
import net.minecraft.item.Item

class FluidStoragePartItem(type: FluidStorageType) : Item(Properties().group(RS.MAIN_GROUP)) {
    companion object {
        fun getByType(type: FluidStorageType): FluidStoragePartItem {
            return when (type) {
                SIXTY_FOUR_K -> RSItems.SIXTY_FOUR_K_FLUID_STORAGE_PART
                TWO_HUNDRED_FIFTY_SIX_K -> RSItems.TWO_HUNDRED_FIFTY_SIX_K_FLUID_STORAGE_PART
                THOUSAND_TWENTY_FOUR_K -> RSItems.THOUSAND_TWENTY_FOUR_K_FLUID_STORAGE_PART
                FOUR_THOUSAND_NINETY_SIX_K -> RSItems.FOUR_THOUSAND_NINETY_SIX_K_FLUID_STORAGE_PART
                else -> throw IllegalArgumentException("Cannot get fluid storage part of $type")
            }
        }
    }

    init {
        this.setRegistryName(RS.ID, type.getName().toString() + "_fluid_storage_part")
    }
}